packages/database/src/migrations/20210623041504-AddNERGERLESMIS-modifies-data.js

```diff
@@ -0,0 +1,231 @@
+    config: reportConfig,
+    permission_group_id: permissionGroupId,
+  });
+
+  // insert dashboard item
+  const dashboardItemId = generateId();
+  await insertObject(db, 'dashboard_item', {
+    id: dashboardItemId,
+    code,
+    config: frontEndConfig,
+    report_code: code,
+  });
+
+  // insert relation record connecting dashboard item to dashboard
+  await addItemToDashboard(db, { code, dashboardCode, entityTypes, projectCodes, permissionGroup });
+};
+
+const addItemToDashboard = async (
+  db,
+  { code, dashboardCode, permissionGroup, entityTypes, projectCodes },
+) => {
+  const dashboardItemId = (
+    await findSingleRecord(db, `SELECT id FROM dashboard_item WHERE code = '${code}'`)
+  ).id;
+  const dashboardId = (
+    await findSingleRecord(db, `SELECT id FROM dashboard WHERE code = '${dashboardCode}'`)
+  ).id;
+  const maxSortOrder = (
+    await findSingleRecord(
+      db,
+      `SELECT max(sort_order) as max_sort_order FROM dashboard_relation WHERE dashboard_id = '${dashboardId}';`,
+    )
+  ).max_sort_order;
+  await insertObject(db, 'dashboard_relation', {
+    id: generateId(),
+    dashboard_id: dashboardId,
+    child_id: dashboardItemId,
+    entity_types: `{${entityTypes.join(', ')}}`,
+    project_codes: `{${projectCodes.join(', ')}}`,
+    permission_groups: `{${permissionGroup}}`,
+    sort_order: maxSortOrder + 1,
+  });
+};
+
+const removeDashboardItemAndReport = async (db, code) => {
+  await db.runSql(`DELETE FROM dashboard_item WHERE code = '${code}';`); // delete cascades to dashboard_relation
+  await db.runSql(`DELETE FROM report WHERE code = '${code}';`);
+};
+
+exports.up = async function (db) {
+  const baseConfigString = JSON.stringify({
+    code: CODE,
+    reportConfig: REPORT_CONFIG,
+    frontEndConfig: FRONT_END_CONFIG,
+    permissionGroup: 'LESMIS Public',
+    dashboardCode: 'LA_Student_Enrolment',
+    entityTypes: ['tupaia_entity_level'],
+    projectCodes: ['laos_schools'],
+  });
+
+  const configs = LESMIS_ENTITY_LEVELS.flatMap((lesmisEntityLevel, i) => {
+    const tupaiaEntityLevel = TUPAIA_ENTITY_LEVELS[i];
+
+    // replace entity level placeholders in config
+    const entityLevelsReplaced = baseConfigString
+      .replace(/lesmis_entity_level/g, lesmisEntityLevel)
+      .replace(/tupaia_entity_level/g, tupaiaEntityLevel);
+
+    // replace enrolment rate placeholders in config
+    return ENROLMENT_RATE_ACRONYMS.map((acronym, j) => {
+      const title = ENROLMENT_RATE_TITLES[j];
+      return entityLevelsReplaced
+        .replace(/enrolment_rate_acronym/g, acronym)
+        .replace(/enrolment_rate_title/g, title);
+    });
+  }).map(configString => JSON.parse(configString));
+
+  for (const config of configs) {
+    await addNewDashboardItemAndReport(db, config);
+  }
+};
+
+exports.down = async function (db) {
+  for (const lesmisEntityLevel of LESMIS_ENTITY_LEVELS) {
+    await removeDashboardItemAndReport(db, CODE.replace('lesmis_entity_level', lesmisEntityLevel));
```

Comment:
I don't think this down would work, `CODE` also has the acronym so you'd need to replace that too

```
const CODE = 'LESMIS_enrolment_rate_acronym_lesmis_entity_level';
```

Reply:
Good catch! Updated